Fix CI/CD

## Fix issues with CI/CD so that program development can run smoothly and automatically, and hide secret keys in .gitignore.

- [ ] Converted from single-file script to proper package structure with src/cookiespy/ layout
- [ ] Added comprehensive test suite with pytest and GitHub Actions CI/CD workflows
- [ ] Enhanced CLI functionality with better error handling and URL validation

This will be fixed soon so that it can be released to PyPI @fdhliakbar 

Publish/Run Test
```bash
Run pytest -v
  pytest -v
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.13.6/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.13.6/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.6/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.6/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.6/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.13.6/x64/lib
============================= test session starts ==============================
platform linux -- Python 3.13.6, pytest-8.4.1, pluggy-1.6.0 -- /opt/hostedtoolcache/Python/3.13.6/x64/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/CookieSpy/CookieSpy
configfile: pyproject.toml
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
______________________ ERROR collecting tests/test_cli.py ______________________
ImportError while importing test module '/home/runner/work/CookieSpy/CookieSpy/tests/test_cli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.13.6/x64/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/test_cli.py:5: in <module>
    from cookiespy.cli import fetch_cookies, validate_url
/opt/hostedtoolcache/Python/3.13.6/x64/lib/python3.13/site-packages/cookiespy/__init__.py:1: in <module>
    from .cli import main, fetch_cookies, validate_url
/opt/hostedtoolcache/Python/3.13.6/x64/lib/python3.13/site-packages/cookiespy/cli.py:3: in <module>
    from rich.console import Console
E   ModuleNotFoundError: No module named 'rich'
=========================== short test summary info ============================
ERROR tests/test_cli.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.24s ===============================
Error: Process completed with exit code 2.
```
Damn bro.

--- a/.github/workflows/ci-release.yml
+++ b/.github/workflows/ci-release.yml
@@ -21,11 +21,14 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
+          pip install -e .
           pip install pytest ruff
       - name: Lint (ruff)
         run: ruff check .
       - name: Test (pytest)
-        run: pytest -q
+        run: |
+          echo "PYTHONPATH=$PWD/src" >> $GITHUB_ENV
+          pytest -v --import-mode=importlib
 
   build-wheel:
     if: startsWith(github.ref, 'refs/tags/v')

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -29,7 +29,7 @@ jobs:
           pip install pytest ruff
       - name: Lint (ruff)
         run: ruff check .
+      - name: Set PYTHONPATH
+        run: echo "PYTHONPATH=$PWD/src" >> $GITHUB_ENV
       - name: Run tests
-        run: |
-          export PYTHONPATH=src
-          pytest -v --import-mode=importlib
+        run: pytest -v --import-mode=importlib

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -30,4 +30,6 @@ jobs:
       - name: Lint (ruff)
         run: ruff check .
       - name: Run tests
-        run: pytest -v
+        run: |
+          export PYTHONPATH=src
+          pytest -v --import-mode=importlib

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -25,6 +25,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
+          pip install .
           pip install pytest ruff
       - name: Lint (ruff)
         run: ruff check .

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -1,19 +1,32 @@
+
 name: Run Tests
 
 on:
   push:
+    branches:
+      - development
   pull_request:
+    branches:
+      - development
 
 jobs:
   test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.10', '3.11', '3.12' ]
     steps:
       - uses: actions/checkout@v4
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.13"
+          python-version: ${{ matrix.python-version }}
       - name: Install dependencies
-        run: pip install . pytest
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest ruff
+      - name: Lint (ruff)
+        run: ruff check .
       - name: Run tests
         run: pytest -v

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -25,7 +25,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
-          pip install .
+          pip install -e .
           pip install pytest ruff
       - name: Lint (ruff)
         run: ruff check .

